pkg/controllers/awsloadbalancercontroller/ingressclass_test.go

```diff
@@ -0,0 +1,106 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"testing"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+func TestDesiredIngressClass(t *testing.T) {
+	ic := desiredIngressClass("test")
+	if ic.Name != "test" {
+		t.Errorf("unexpected name in desired ingress class, expected %q, got %q", "test", ic.Name)
+	}
+	if ic.Spec.Controller != albIngressClassController {
+		t.Errorf("unexpected controller in desired ingress class, expected %q, got %q", albIngressClassController, ic.Spec.Controller)
+	}
+}
+
+func TestEnsureIngressClass(t *testing.T) {
+	for _, tc := range []struct {
+		name                 string
+		existingIngressClass *networkingv1.IngressClass
+		ingressClassName     string
+		deletedIngressClass  bool
```

Comment:
I agree. Personally I prefer the second approach because the event based test does not ensure that the reconciled resource as actually as they are supposed to be after the reconcile.